server/konnectors/edf.coffee

```diff
@@ -0,0 +1,1358 @@
+                    'message:numeroBp': entries.client.clientId
+                    'message:numeroContrat': contract.number
+
+        edfRequestPost path, body, (err, result) ->
+            return callback err if err
+
+            try
+                unless "ns:corpsSortie" of result["ns:msgReponse"]
+                    K.logger.info "No histoConsos to fetch"
+                    return callback null, []
+
+                consoElems = result["ns:msgReponse"]["ns:corpsSortie"][0]\
+                    ["ns:listeHistoDeConso"]
+
+                res = consoElems.map (consoElem) ->
+                    doc =
+                        contractNumber: contract.number
+                        billNumber: getF consoElem, 'ns:numeroFacture'
+                        start: getF consoElem, 'ns:dateDebut'
+                        end: getF consoElem, 'ns:dateFin'
+                        value: getF consoElem, 'ns:listeConsommation'
+                        , 'ns:valeur'
+                        # unit: getF conso, 'ns:listeConsommation', 'ns:cadran'
+                        statementType: getF consoElem, 'ns:typeReleve'
+                        statementCategory: getF consoElem, 'ns:categorieReleve'
+                        statementReason: getF consoElem, 'ns:motifReleve'
+                        docTypeVersion: getDocTypeVersion()
+
+                    return doc
+
+                cb null, res
+            catch e
+                K.logger.error "While fetching visualiserHistoConso."
+                K.logger.error e
+                cb e
+
+    , (err, results) ->
+        return callback err if err
+
+        entries.consumptionStatements = results.reduce (agg, result) ->
+            return agg.concat result
+        , []
+
+        K.logger.info "Fetched #{entries.consumptionStatements.length}"+
+            " consumptionStatements"
+        callback()
+
+
+fetchPDF = (token, client, billNumber, callback) ->
+    K.logger.info "fetchPDF"
+
+    path = '/ws/recupererDocumentContractuelGet_rest_V1-0/invoke'
+    body =
+        'ns:msgRequete':
+            '$':
+                'xmlns:dicoPAS': "http://www.edf.fr/commerce/passerelle/" +
+                "commun/v2/dico"
+                'xmlns:dico': "http://www.edf.fr/psc/pscmaxsd/commun/v1/dico"
+                'xmlns:ns': "http://www.edf.fr/commerce/passerelle/psc/" +
+                "recupererDocumentContractuelGet/service/v1"
+                'xmlns:ent': "http://www.edf.fr/commerce/passerelle/commun/" +
+                "v2/entete"
+
+            'ns:jeton': token
+            'ns:options': [
+                'ns:cle': 'id'
+                'ns:valeur': 'pscedfmoi'
+            ,
+                'ns:cle': 2
+                'ns:valeur': client.clientId
+            ,
+                'ns:cle': 4
+                'ns:valeur': billNumber
+            ,
+                'ns:cle': 6
+                'ns:valeur': 'Facture'
+            ]
+
+    edfRequestPost path, body, (err, result) ->
```

Comment:
Sadly no, because I have to use the specific token and request format for the request.